lib/private/Files/SetupManager.php

```diff
@@ -81,8 +98,20 @@ public function __construct(
 		$this->eventDispatcher = $eventDispatcher;
 		$this->userMountCache = $userMountCache;
 		$this->lockdownManager = $lockdownManager;
+		$this->logger = $logger;
 		$this->userSession = $userSession;
+		$this->cache = $cacheFactory->createDistributed('setupmanager::');
 		$this->listeningForProviders = false;
+
+		$this->setupListeners();
+	}
+
+	private function isSetupStarted(IUser $user): bool {
+		return in_array($user->getUID(), $this->setupUsers, true);
```

Comment:
How often are those used? Would it make sense to store uids as keys in the array to speed up look up and avoid in_array?
 ```suggestion
		return $this->setupUsers[$user->getUID()] ?? false;
```